### PR TITLE
SONAR-6261 Move events.resource_id to events.component_uuid

### PR DESCRIPTION
**Pull request by julienlancelot**
SONAR-6261 Move events.resource_id to events.component_uuid

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/computation/step/PersistEventsStep.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+  private void deletePreviousEventsHavingSameVersion(DbSession session, BatchReport.Component component){
+    for (EventDto dto : dbClient.eventDao().selectByComponentUuid(session, component.getUuid())) {
+      if (dto.getCategory().equals(EventDto.CATEGORY_VERSION) && dto.getName().equals(component.getVersion())) {
+        dbClient.eventDao().delete(session, dto.getId());
+      }
+    }
+  }
+
+  private EventDto createBaseEvent(BatchReport.Component component, Long analysisDate) {
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/computation/step/PersistEventsStep.java]: rename to newBaseEvent or rename createVersionEvent to saveVersionEvent – my point is to have similar action with a similar name. One is persisting in db this other is creating a dto

File: server/sonar-web/src/main/webapp/WEB-INF/app/controllers/project_controller.rb [hunk 2]
```diff
@@ -385,7 +385,7 @@ def update_event
       flash[:notice] = message('project_history.event_updated')
     end
 
-    redirect_to :action => 'history', :id => event.resource_id
+    redirect_to :action => 'history', :id => event.resource.id
```
Comment (teryk) [inline server/sonar-web/src/main/webapp/WEB-INF/app/controllers/project_controller.rb]: how does it work ? how the event object contains the resource objet ?

File: sonar-batch/src/main/java/org/sonar/batch/deprecated/components/PastSnapshotFinderByPreviousVersion.java [hunk 3]
```diff
@@ -43,8 +42,8 @@ public PastSnapshot findByPreviousVersion(Snapshot projectSnapshot) {
     String currentVersion = projectSnapshot.getVersion();
     Integer resourceId = projectSnapshot.getResourceId();
 
-    String hql = "from " + Event.class.getSimpleName() +
-      " where name<>:version AND category='Version' AND resourceId=:resourceId ORDER BY date DESC";
+    String hql = "select e from " + Event.class.getSimpleName() + " e " +
+      " join e.resource component where e.name<>:version AND e.category='Version' AND component.id=:resourceId ORDER BY e.date DESC";
```
Comment (teryk) [inline sonar-batch/src/main/java/org/sonar/batch/deprecated/components/PastSnapshotFinderByPreviousVersion.java]: why not replace e.resource component with e.resource c ?

File: sonar-core/src/main/java/org/sonar/core/event/EventDto.java [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+  public String getData() {
+    return data;
+  }
+
+  public EventDto setData(@Nullable String data) {
+    this.data = data;
+    return this;
+  }
+
+  public Long getDate() {
```
Comment (teryk) [inline sonar-core/src/main/java/org/sonar/core/event/EventDto.java]: I'd prefer a java.util.Date. A Long is misleading, what does it represent ?

File: server/sonar-server/src/main/java/org/sonar/server/computation/step/PersistEventsStep.java (hunk 1 above)
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/computation/step/PersistEventsStep.java]: better, name toEventDto or something similar

File: sonar-core/src/main/java/org/sonar/core/event/EventDto.java (hunk 4 above)
Comment (julienlancelot) [inline sonar-core/src/main/java/org/sonar/core/event/EventDto.java]: I prefer long, and add a helper that return a Date if needed. As we're now using long for every dates that we're using, using long everywhere seems good to me.
The only place to use a Date is in WS and in E/S.

File: server/sonar-server/src/main/java/org/sonar/server/computation/step/PersistEventsStep.java (hunk 1 above)
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/computation/step/PersistEventsStep.java]: Ok, I'll rename createVersionEvent to saveVersionEvent and rename createBaseEvent to newBaseEvent

File: server/sonar-web/src/main/webapp/WEB-INF/app/controllers/project_controller.rb (hunk 2 above)
Comment (julienlancelot) [inline server/sonar-web/src/main/webapp/WEB-INF/app/controllers/project_controller.rb]: It's using a join, See event.rb -> belongs_to :resource.

File: sonar-core/src/main/java/org/sonar/core/event/EventDto.java (hunk 4 above)
Comment (teryk) [inline sonar-core/src/main/java/org/sonar/core/event/EventDto.java]: then I want us to agree on that, overall consistency > my own preference.

Comment (julienlancelot) [inline sonar-core/src/main/java/org/sonar/core/event/EventDto.java]: Ok, let's have a talk about this

**Comment by julienlancelot** [comment]
@teryk Is it good for you ?

**Comment by teryk** [comment]
LGTM, I really want to have this talk about dates : )

**Pull request closed without merging**